x-pack/plugins/ingest_manager/server/services/agents/enroll.test.ts

```diff
@@ -5,62 +5,52 @@
  */
 
 import { validateAgentVersion } from './enroll';
-import { appContextService } from '../app_context';
-import { IngestManagerAppContext } from '../../plugin';
 
 describe('validateAgentVersion', () => {
   it('should throw with agent > kibana version', () => {
-    appContextService.start(({
-      kibanaVersion: '8.0.0',
-    } as unknown) as IngestManagerAppContext);
-    expect(() =>
-      validateAgentVersion({
-        local: { elastic: { agent: { version: '8.8.0' } } },
-        userProvided: {},
-      })
-    ).toThrowError(/Agent version is not compatible with kibana version/);
+    expect(() => validateAgentVersion('8.8.0', '8.0.0')).toThrowError('not compatible');
   });
   it('should work with agent < kibana version', () => {
-    appContextService.start(({
-      kibanaVersion: '8.0.0',
-    } as unknown) as IngestManagerAppContext);
-    validateAgentVersion({ local: { elastic: { agent: { version: '7.8.0' } } }, userProvided: {} });
+    validateAgentVersion('7.8.0', '8.0.0');
   });
 
   it('should work with agent = kibana version', () => {
-    appContextService.start(({
-      kibanaVersion: '8.0.0',
-    } as unknown) as IngestManagerAppContext);
-    validateAgentVersion({ local: { elastic: { agent: { version: '8.0.0' } } }, userProvided: {} });
+    validateAgentVersion('8.0.0', '8.0.0');
   });
 
   it('should work with SNAPSHOT version', () => {
-    appContextService.start(({
-      kibanaVersion: '8.0.0-SNAPSHOT',
-    } as unknown) as IngestManagerAppContext);
-    validateAgentVersion({
-      local: { elastic: { agent: { version: '8.0.0-SNAPSHOT' } } },
-      userProvided: {},
-    });
+    validateAgentVersion('8.0.0-SNAPSHOT', '8.0.0-SNAPSHOT');
   });
 
   it('should work with a agent using SNAPSHOT version', () => {
-    appContextService.start(({
-      kibanaVersion: '7.8.0',
-    } as unknown) as IngestManagerAppContext);
-    validateAgentVersion({
-      local: { elastic: { agent: { version: '7.8.0-SNAPSHOT' } } },
-      userProvided: {},
-    });
+    validateAgentVersion('7.8.0-SNAPSHOT', '7.8.0');
   });
 
   it('should work with a kibana using SNAPSHOT version', () => {
-    appContextService.start(({
-      kibanaVersion: '7.8.0-SNAPSHOT',
-    } as unknown) as IngestManagerAppContext);
-    validateAgentVersion({
-      local: { elastic: { agent: { version: '7.8.0' } } },
-      userProvided: {},
-    });
+    validateAgentVersion('7.8.0', '7.8.0-SNAPSHOT');
+  });
+
+  it('very close versions, e.g. patch/prerelease - all combos should work', () => {
```

Comment:
cc: @EricDavisX wdyt of these tests? They are basically `validate(agent_version, kibana_version)`

Any other combinations you'd like added to the tests?